spec/requests/api/v2/tube_from_tube_creations_spec.rb

```diff
@@ -0,0 +1,297 @@
+
+        it 'associates the parent with the new record' do
+          new_record = model_class.last
+          expect(new_record.parent).to eq(parent)
+        end
+
+        it 'associates the user with the new record' do
+          new_record = model_class.last
+          expect(new_record.user).to eq(user)
+        end
+
+        it 'generated a child with valid attributes' do
+          new_record = model_class.last
+          expect(new_record.child.purpose).to eq(child_purpose)
+        end
+      end
+
+      context 'with complete attributes' do
+        let(:payload) do
+          {
+            data: {
+              type: resource_type,
+              attributes: {
+                child_purpose_uuid: child_purpose.uuid,
+                parent_uuid: parent.uuid,
+                user_uuid: user.uuid
+              }
+            }
+          }
+        end
+
+        it_behaves_like 'a valid request'
+      end
+
+      context 'with complete relationships' do
+        let(:payload) do
+          {
+            data: {
+              type: resource_type,
+              relationships: {
+                child_purpose: child_purpose_relationship,
+                parent: parent_relationship,
+                user: user_relationship
+              }
+            }
+          }
+        end
+
+        it_behaves_like 'a valid request'
+      end
+
+      context 'with conflicting relationships' do
+        let(:other_parent) { create(:tube, prefix: 'PT') }
+        let(:other_user) { create(:user) }
+        let(:payload) do
+          {
+            data: {
+              type: resource_type,
+              attributes: {
+                child_purpose_uuid: child_purpose.uuid,
+                parent_uuid: other_parent.uuid,
+                user_uuid: other_user.uuid
+              },
+              relationships: {
+                child_purpose: child_purpose_relationship,
+                parent: parent_relationship,
+                user: user_relationship
+              }
+            }
+          }
+        end
+
+        # This test should pass because the relationships are preferred over the attributes.
+        it_behaves_like 'a valid request'
+      end
+    end
+
+    context 'with a read-only attribute in the payload' do
+      context 'with uuid' do
+        let(:disallowed_value) { 'uuid' }
+        let(:payload) { { data: { type: resource_type, attributes: { uuid: '111111-2222-3333-4444-555555666666' } } } }
+
+        it_behaves_like 'a POST request with a disallowed value'
+      end
+    end
+
+    context 'with a read-only relationship in the payload' do
+      context 'with child' do
+        let(:disallowed_value) { 'child' }
+        let(:payload) do
+          { data: { type: resource_type, relationships: { child: { data: { id: '1', type: 'plates' } } } } }
+        end
+
+        it_behaves_like 'a POST request with a disallowed value'
+      end
+    end
+
+    context 'without a required relationship' do
```

Comment:
is child_purpose also a required relationship?

Reply:
Weirdly, no.  The model seems happy without it.